emission/analysis/intake/segmentation/restart_checking.py

```diff
@@ -9,36 +9,40 @@
 import emission.core.wrapper.transition as ecwt
 import emission.storage.timeseries.timequery as estt
 
-def is_tracking_restarted_in_range(start_ts, end_ts, timeseries):
+def is_tracking_restarted_in_range(start_ts, end_ts, transition_df):
     """
     Check to see if tracing was restarted between the times specified
     :param start_ts: the start of the time range to check
     :param end_ts: the end of the time range to check
     :param timeseries: the timeseries to use for checking
     :return:
     """
-    import emission.storage.timeseries.timequery as estt
+    transition_df_start_idx=transition_df.ts.searchsorted(start_ts,side='left')    
+    transition_df_end_idx=transition_df.ts.searchsorted(end_ts,side='right')
+    transition_df_for_current=transition_df.iloc[transition_df_start_idx:transition_df_end_idx]
```

Comment:
Is there a reason you are using this instead of something like

```
    transition_df_for_current=transition_df[transition_df.ts >= start_ts && transition_df.ts <= end_ts]
```

or 

```
    transition_df_for_current=transition_df.query('ts >= start_ts & .ts <= end_ts')
```

Reply:
First was the performance reason, O(log (n)) here vs O (n) in others. 
Second, in case of query(), it  creates a boolean series that marks rows to keep or discard, which happens internally, which increases temporary memory usage. For very large DataFrames, this can be an issue.

can use this one ,if log(n) vs O(n) is not an issue : 

   ` transition_df_for_current=transition_df[transition_df.ts >= start_ts && transition_df.ts <= end_ts]`